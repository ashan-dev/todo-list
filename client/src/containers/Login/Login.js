const handleLogin = (event, history, loginAction) => {
  event.preventDefault();
  const { email, password } = event.target.elements;
  try {
    loginAction(
      {
        email: email.value,
        password: password.value,
      },
      () => {
        history.push("/");
      }
    );
  } catch (error) {
    alert(error);
  }
};
export { handleLogin };
