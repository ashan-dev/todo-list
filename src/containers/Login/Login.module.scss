.container {
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  & .loginContainer {
    width: 400px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    & .title {
      font-family: "Roboto-Bold";
      color: #000000;
      font-size: 26px;
    }
    & .loginFormContainer {  
      padding: 20px;
      border-radius: 10px;
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
      box-shadow: 0px 2px 5px 1px #888888;
      & .inputField {
        border-radius: 4px;
        border: 0.5px solid rgba(0, 0, 0, 0.2);
        width: 350px;
        margin-bottom: 10px;
        padding: 10px;
      }
      & .loginBtnContainer {
        background-color: #8fd5ff;
        border: none;
        border-radius: 6px;
        padding: 5px 10px;
        font-family: "Roboto-Medium";
        color: rgba(0, 0, 0, 0.9);
        font-size: 16px;
        margin-top: 10px;
      }
    }
  }
}
