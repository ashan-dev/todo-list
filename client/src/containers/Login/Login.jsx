import React from "react";
import { Redirect } from "react-router";
import { connect } from "react-redux";
import { useHistory } from 'react-router-dom';
import { handleLogin } from "./Login.js";
import { loginAction } from "../../redux/actions/user-action";
import styles from "./Login.module.scss";

const Login = ({ login, loggedInUserData }) => {
  const history = useHistory()
  if (loggedInUserData && Object.keys(loggedInUserData).length > 0) {
    return <Redirect to="/" />;
  } else {
    return (
      <div className={styles.container}>
        <div className={styles.loginContainer}>
          <p className={styles.title}>To-do List</p>

          <form
            className={styles.loginFormContainer}
            onSubmit={(event) => {
              handleLogin(event, history, login);
            }}
          >
            <div>
              <input className={styles.inputField} name="email" type="email" placeholder="Email" />
            </div>
            <div>
              <input className={styles.inputField} name="password" type="password" placeholder="Password" />
            </div>
            <button className={styles.loginBtnContainer} type="submit">Log in</button>
          </form>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state) => ({
  loggedInUserData: state.userState.userData,
});

const mapDispatchToProps = (dispatch) => ({
  login: (loginData, callback) => dispatch(loginAction(loginData, callback)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
