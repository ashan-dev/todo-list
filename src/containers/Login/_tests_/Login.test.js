import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Login from '../Login.jsx';

const mockStore = configureMockStore();
const store = mockStore({
  userState: {
    userData: null,
  },
});

test('Login renders correctly', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Login />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
