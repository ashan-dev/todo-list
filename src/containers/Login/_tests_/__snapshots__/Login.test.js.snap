// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="loginContainer"
  >
    <p
      className="title"
    >
      To-do List
    </p>
    <form
      className="loginFormContainer"
      onSubmit={[Function]}
    >
      <div>
        <input
          className="inputField"
          name="email"
          placeholder="Email"
          type="email"
        />
      </div>
      <div>
        <input
          className="inputField"
          name="password"
          placeholder="Password"
          type="password"
        />
      </div>
      <button
        className="loginBtnContainer"
        type="submit"
      >
        Log in
      </button>
    </form>
  </div>
</div>
`;
